api/apps/api/src/modules/access-control/projects-acl/project-acl.service.ts

```diff
@@ -79,4 +80,56 @@ export class ProjectAclService implements ProjectAccessControl {
 
     return intersection(roles, this.canViewSolutionRoles).length > 0;
   }
+
+  async checkUserIsOwner(userId: string, projectId: string): Promise<boolean> {
+    const userIsProjectOwner = await this.roles.findOne({
+      where: {
+        projectId,
+        userId,
+        roleName: Roles.project_owner,
+      },
+    });
+
+    return !!userIsProjectOwner;
+  }
+
+  async findUsersInProject(
+    projectId: string,
+  ): Promise<UsersProjectsApiEntity[]> {
+    const usersInProject = await this.roles.find({
+      where: { projectId },
+      select: ['roleName'],
+      relations: ['user'],
+    });
+    return usersInProject;
+  }
+
+  async updateUserInProject(
+    projectId: string,
+    updateUserInProjectDto: UpdateUserRoleinProject,
+  ): Promise<void> {
+    const { userId, roleName } = updateUserInProjectDto;
+    const existingUserInProject = await this.roles.findOne({
+      where: {
+        projectId,
+        userId,
+      },
+    });
+    if (!existingUserInProject) {
+      await this.roles.save({
+        projectId,
+        userId,
+        roleName,
+      });
+    } else {
+      await this.roles.update({ projectId, userId }, { roleName });
+    }
```

Comment:
It may be rare but imagine two parallel requests hitting:
- one to delete
- second to update

so, it may happen that the `update` first part (getting existingUserInProject) executes, then the delete happens, then we are back to update user... which may not work as we expect.

Most suitable would be to wrap find/update within transaction, locking the row till finished.